client/my-sites/domains/controller.jsx

```diff
@@ -256,7 +256,14 @@ const jetpackNoDomainsWarning = ( context, next ) => {
 					path={ context.path.indexOf( '/domains/add' ) === 0 ? '/domains/add' : '/domains/manage' }
 					title="My Sites > Domains > No Domains On Jetpack"
 				/>
-				<JetpackManageErrorPage template="noDomainsOnJetpack" siteId={ selectedSite.ID } />
+				<EmptyContent
+					title={ translate( 'Domains are not available for this site.' ) }
+					line={ translate(
+						'You can only purchase domains for sites hosted on WordPress.com at this time.'
+					) }
+					action={ translate( 'View Plans' ) }
+					actionURL={ '/plans/' + ( selectedSite.siteSlug || '' ) }
```

Comment:
I'm not seeing my selected site slug in the href for the **View Plans** button. After poking around in the debugger, I think this will do the trick:

```suggestion
					actionURL={ '/plans/' + ( selectedSite.slug || '' ) }
```